src/smartpeak/include/SmartPeak/core/Parameters.h

```diff
@@ -255,6 +262,11 @@ namespace SmartPeak
 
     Parameter* findParameter(const std::string& parameter);
     const std::string& getFunctionName() const { return function_name_; };
+    
+    /**
+      @brief return a vector of the currently loaded parameters
+    */
+    std::vector<Parameter> getParameters() const { return parameters_; };
```

Comment:
Do we need that? the FunctionParameter is iterable.

Reply:
that's essential to look up the right parameter